### PR TITLE
Add sharding package and update validators

### PR DESCRIPTION
**Pull request by lkatalin**
#### Summary
Commit 1: Add sharding package. I moved this to `pkg/` so it can be accessed by both the server and cli as needed, but if this is no the right place, I can move it.
Commit 2: Update UUID validators to accept both current 64-char UUID and longer 80-char FullID defined in sharding package.

#### Ticket Link

Related to #487 
(More work to be done before the issue is closed)

### CODE REVIEW

**Review by cpanato — COMMENTED** [review]
lgtm

File: pkg/sharding/sharding.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		err := fmt.Errorf("invalid uuid len: %v", len(uuid))
+		return CreateEmptyFullID(), err
+	}
+
+	return FullID{
+		TreeID: treeid,
+		UUID:   uuid}, nil
+}
+
+func CreateEmptyFullID() FullID {
```
Comment (bobcallaway) [inline pkg/sharding/sharding.go]: is this intent for this to be public or just a internal helper function?

File: pkg/sharding/sharding.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+var dummy = api.LogRanges{
+	Ranges: []api.LogRange{},
+}
+
+type FullID struct {
+	TreeID string
+	UUID   string
+}
+
+func CreateFullID(treeid string, uuid string) (FullID, error) {
```
Comment (bobcallaway) [inline pkg/sharding/sharding.go:53]: do you need any sort of validation in this method to ensure the arguments are valid hex strings?

File: cmd/rekor-cli/app/pflags.go [hunk 3]
```diff
@@ -190,6 +191,22 @@ func validateFileOrURL(v string) error {
...
+// validateID ensures the ID is either a FullID (TreeID + UUID) or a UUID
+func validateID(v string) error {
+	if len(v) != sharding.FullIDHexStringLen && len(v) != sharding.UUIDHexStringLen {
+		return fmt.Errorf("ID len error, expected %v (FullID) or %v (UUID) but got len %v for ID %v", sharding.FullIDHexStringLen, sharding.UUIDHexStringLen, len(v), v)
+	}
+
+	IDStringValidatorFunc := validateString("required,hexadecimal")
+	IDErr := IDStringValidatorFunc(v)
+
+	if IDErr != nil {
```
Comment (bobcallaway) [inline cmd/rekor-cli/app/pflags.go]:
```suggestion
	if err := validateString("required,hexadecimal")(v); err != nil {
```

Comment (lkatalin) [reply cmd/rekor-cli/app/pflags.go]: Nice, thank you! I'm making this change.

File: pkg/sharding/sharding.go (hunk 1 above)
Comment (lkatalin) [reply pkg/sharding/sharding.go]: Good catch, I'm still getting used to go's use of caps to mean public. I suppose for now it should be internal so I'll update it.

File: pkg/sharding/sharding.go (hunk 2 above)
Comment (lkatalin) [reply pkg/sharding/sharding.go:53]: Thanks, added!

**Comment by lkatalin** [comment]
Thanks for the helpful comments, @bobcallaway . I've made some updates. Also I'm not sure what we should call the `TreeID + UUID`. I've been using `FullID`, but maybe there is something more descriptive?

**Review by bobcallaway — APPROVED** [review]
lgtm, thanks!

**Comment by bobcallaway** [comment]
> Thanks for the helpful comments, @bobcallaway . I've made some updates. Also I'm not sure what we should call the `TreeID + UUID`. I've been using `FullID`, but maybe there is something more descriptive?

`EntryID`?

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
Nice job!

**Pull request merged**